arch/sim/src/cmake/Toolchain.cmake

```diff
@@ -18,6 +18,17 @@
 #
 # ##############################################################################
 
+add_compile_options(
+  -U_AIX
+  -U_WIN32
+  -U__APPLE__
+  -U__FreeBSD__
+  -U__NetBSD__
+  -U__linux__
+  -U__sun__
+  -U__unix__
+  -U__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__)
```

Comment:
unify the changes
https://github.com/apache/nuttx/blob/master/arch/sim/CMakeLists.txt#L68-L93

Reply:
@anchao OK, let's unify the SIM option into Toolchain file and remove setting in SIM board cmakelists.txt

Reply:
Done